tests/e2e/performance/csi_tests/test_auto_reclaim_space_multi_clones.py

```diff
@@ -0,0 +1,243 @@
+
+        # Getting the total Storage capacity
+        ceph_capacity = int(self.ceph_cluster.get_ceph_capacity())
+        # Getting the free Storage capacity
+        ceph_free_capacity = int(self.ceph_cluster.get_ceph_free_capacity())
+        # Use 70% of the free storage capacity in the test
+        capacity_to_use = int(ceph_free_capacity * 0.7)
+
+        # since we do not want to use more then 65%, we add 35% to the needed
+        # capacity, and minimum PVC size is 1 GiB
+        need_capacity = int((self.num_of_clones + 2) * 1.35)
+        # Test will run only on system with enough capacity
+        if capacity_to_use < need_capacity:
+            err_msg = (
+                f"The system has only {ceph_capacity} GiB, "
+                f"Of which {ceph_free_capacity} GiB is free, "
+                f"we want to use  {capacity_to_use} GiB, "
+                f"and we need {need_capacity} GiB to run the test"
+            )
+            logger.error(err_msg)
+            raise StorageNotSufficientException(err_msg)
+
+        # Calculating the PVC size in GiB
+        self.pvc_size = int(capacity_to_use / (self.num_of_clones + 2))
+
+        logger.info(
+            f"Total capacity size is : {ceph_capacity} GiB, "
+            f"Free capacity size is : {ceph_free_capacity} GiB, "
+            f"Going to use {need_capacity} GiB, "
+            f"With {self.num_of_clones} clones to {self.pvc_size} GiB PVC. "
+        )
+
+        namespace = f"openshift-{uuid4().hex}"
+        self.namespace = namespace
+        result = run_oc_command(cmd=f"create namespace {self.namespace}")
+        assert ERRMSG not in result[0], (
+            f"Failed to create namespace with name {namespace}" f"got result: {result}"
+        )
+        logger.info(f"Namespace {namespace} created")
+
+        self.cloned_obj_list = []
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Cleanup the test environment
+        """
+
+        def finalizer():
+            logger.info("Starting the test cleanup")
+
+            # Delete the created clones
+            try:
+                for clone in self.cloned_obj_list:
+                    performance_lib.run_oc_command(
+                        cmd=f"delete pvc {clone}", namespace=self.namespace
+                    )
+            except Exception:
+                logger.warning("Clones were not deleted")
+
+            # Delete the test PVC
+            self.cleanup_testing_pvc()
+
+            # Delete the test StorageClass
+            try:
+                logger.info(f"Deleting the test StorageClass : {self.sc_obj.name}")
+                self.sc_obj.delete()
+                logger.info("Wait until the SC is deleted.")
+                self.sc_obj.ocp.wait_for_delete(resource_name=self.sc_obj.name)
+            except Exception as ex:
+                logger.error(f"Can not delete the test sc : {ex}")
+
+            # Delete the test storage pool
```

Comment:
again same possibility like my above comment, in-case if Clone and SC deletion fails but storage pool deletion is successful then we can't remove those clone, sc, PVC and PVs.
We have to handle it carefully.

Reply:
@ramkiperiy - Please see my previous comment.